samples/caf_sensor_manager/configuration/common/event_proxy_init.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#define MODULE event_proxy_init
+
+#include <ipc/ipc_service.h>
+#include <event_manager_proxy.h>
+#include <caf/events/sensor_data_aggregator_event.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(MODULE);
+
+int init_event_proxy(char *proxy_subscribe_event)
+{
+	const struct device *ipc_instance  = DEVICE_DT_GET(DT_NODELABEL(ipc0));
+	int ret;
+
+	ret = event_manager_proxy_add_remote(ipc_instance);
+	if (ret) {
+		LOG_ERR("Cannot add remote: %d", ret);
+		__ASSERT_NO_MSG(false);
+		return ret;
+	}
+	LOG_INF("Event proxy remote added");
+
+	if (strcmp(proxy_subscribe_event, "sensor_data_aggregator_event") == 0) {
+		ret = EVENT_MANAGER_PROXY_SUBSCRIBE(ipc_instance, sensor_data_aggregator_event);
+	} else if (strcmp(proxy_subscribe_event,
+			"sensor_data_aggregator_release_buffer_event") == 0) {
+		ret = EVENT_MANAGER_PROXY_SUBSCRIBE(ipc_instance,
+						sensor_data_aggregator_release_buffer_event);
+	}
+	if (ret) {
+		LOG_ERR("Cannot register to remote %s: %d", proxy_subscribe_event, ret);
+		__ASSERT_NO_MSG(false);
+		return ret;
+	}
+	LOG_INF("Event proxy %s registered", proxy_subscribe_event);
+
+	ret = event_manager_proxy_start();
+	if (ret) {
+		LOG_ERR("Cannot start event manager proxy: %d", ret);
+		__ASSERT_NO_MSG(false);
+		return ret;
+	}
+	LOG_INF("Event manager proxy started");
+
+	/* Wait for all the remote cores to finish event proxy initialization. */
```

Comment:
this comment is not needed - check that name of the function is self explanatory